llvm/include/llvm/Analysis/TargetTransformInfo.h

```diff
@@ -1486,7 +1493,7 @@ class TargetTransformInfo {
   /// of the extract(nullptr if user is not known before vectorization) and
   /// 'Idx' being the extract lane.
   LLVM_ABI InstructionCost getVectorInstrCost(
-      unsigned Opcode, Type *Val, TTI::TargetCostKind CostKind, unsigned Index,
+      unsigned Opcode, Type *Val, TTI::TargetCostKind CostKind, int Index,
```

Comment:
Do we need to update the TTI hooks to check if LastIndex has been passed, to avoid unexpected results now that we are calling it with a potentially unexpected value -2?

Reply:
I have done that implicitly by changing all the TTI hooks to call `isKnownVectorIndex` when testing if the index is known or not. Any code that previously compared against -1 now calls `isKnownVectorIndex`, which returns false whether the index is -1 or -2 so as far as I can tell the change is NFC for all other targets.

There may be places where target hooks were never checking for -1 and always assuming a known index - I figured in those cases my patch doesn't make things any more broken than they were already.

Reply:
Ah yes I see. For those places, could we assert that we either have `UnknownIndex` or a known index, i.e. no LastIndex, except for RISV which also supports scalable vectors and would be queried with LastIndex?